app/models/tube.rb

```diff
@@ -70,6 +70,11 @@ def self.delegate_to_purpose(*methods)
   has_one :racked_tube, dependent: :destroy
   has_one :tube_rack, through: :racked_tube
 
+  scope :in_column_major_order, lambda {
+    joins(:racked_tube).order('racked_tubes.coordinate ASC')
+  }
+  delegate :coordinate, to: :racked_tube
+
```

Comment:
How will this react if you try to use 'in_column_major_order' on a Tube that doesn't have a Rack?

Reply:
Tested it right now. It returns an empty list of tubes, because join does not find a match.